lib/si-layer-cache/src/disk_cache.rs

```diff
@@ -75,27 +81,42 @@ impl DiskCache {
     }
 
     async fn cleanup(&self) {
+        let mut would_remove = 0;
+        let mut removed = 0;
+        let now = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .expect("unable to get the current time, what does this mean? How could this happen?")
+            .as_millis();
+
         for md in cacache::list_sync(self.write_path.as_ref()).flatten() {
```

Comment:
So, the way this is written, it will occupy an entire tokio worker thread until it has finished.

add a `tokio::task::yield_now().await` here in each iteration to yield execution in between each removal iteration.